backend/schema/corpora_schema_h5ad_implementation.md

```diff
@@ -0,0 +1,78 @@
+# Corpora Schema AnnData Implementation
+
+Corpora datasets are stored in the HDF5-backed AnnData format as written by version 0.7 of the anndata library. One
+goal of using this format is to have all metadata stored in the same file as the data, so cellxgene will have access to
+all Corpora metadata within a single file.
+
+The Corpora schema requirements and definitions for the AnnData `X`, `uns`, `obs`, and `obsm` attributes are below.
+
+
+### Note on types
+The types below are python3 types. Note that a python3 `str` is a sequence of Unicode code points, which is stored
+null-terminated and UTF-8-encoded by anndata.
+
+For succinctness, there is a `link` type used below that is a `dict` {"link_name": `str`, "link_url": `str`}.
+
+## `X`
+
+Corpora does not impose any additional constraints on the `X` data matrix. So it may be sparse or dense and any
```

Comment:
are we following AnnData conventions for layers / raw?  The corpora spec is a bit thin on this, which is something we need to nail down as it is commonly used, and the distinction is important for downstream use cases.

Reply:
I think wrt cellxgene, all we need to say is that cellxgene is going to display X, right?

Beyond that, I think we need to get more assertive the schema about raw being required.

Reply:
When we will be able to support more than one "layer"?  Do you want to build in the concept now, or just say "there is one X, and it should be raw"?

Reply:
I'd rather not be too prescriptive about it at the moment, since they're on the cusp of redoing the API for layers.